test/CMakeLists.txt

```diff
@@ -6,12 +6,11 @@ set_target_properties(gmock_main PROPERTIES FOLDER "Dependencies")
 
 # common things that the tests need
 add_library(valhalla_test
-  test.h test.cc gurka/gurka.h gurka/gurka.cc
```

Comment:
i think all you really need to do here (no move any code around) is to add an `if` here that check is DATA_TOOLS is enabled and if it is then you can also add the gurka sources to this target.

then inside test.h and test.cc you can check add some `ifdef`s to disable the stuff that depends on mjolnir, rather than mixing all the stuff around.

specifically, not all tests that need mjolnir, need gurka, some of them dont use gurka at all, so forcing the mjolnir dependencies into gurka doesnt make sense

Reply:
Maybe would be better to make another utils file with this function, that depends on DATA_TOOLS=ON (mjolnir)?
I think `ifdefs` is not clear solution and can confusing

Reply:
isnt it just as confusing as adding an `if` to cmake? no matter what we do, we have to conditionally include/exclude code. but yeah if you want you can make a new header and new cc file. maybe `test_data.h/cc` or something

Reply:
Yes, `if` in cmake confusing too. That's why I decided to add it to gurka because it looks like a place that builds with `DATA_TOOLS`

Reply:
There is no way to get around it, there needs to be an `if` somewhere, my argument is that its not uncommon or confusing. Please do not move these functions into gurka they arent gurka-specific.

Reply:
In general, if there is at least one test that uses the functions from mjolnir the tests will breaks down.
Maybe separate in by new folder? Something like `test/mjolnir/test.cc`? (the same way you would do it with a gurka)?